release/scan_docker_images.sh

```diff
@@ -2,49 +2,72 @@
 
 source ../_liferay_common.sh
 
-# Function to scan images using twistcli
 function scan_docker_images {
+	local api_url="https://api.eu.prismacloud.io"
+	local console="https://europe-west3.cloud.twistlock.com/eu-1614931"
 	local image_names=("$@") # Capture all arguments as an array
-	local CONSOLE="https://europe-west3.cloud.twistlock.com/eu-1614931"
-	local API_URL="https://api.eu.prismacloud.io"
 
-	# Retrieve credentials from environment variables (assuming Jenkins sets them)
-	PRISMA_ACCESS_KEY=""
-	PRISMA_SECRET=""
+	LIFERAY_PRISMA_ACCESS_KEY=""
+	LIFERAY_PRISMA_SECRET=""
 
-	# Authenticate and retrieve token
-	auth_response=$(curl -s -X POST \
-		-H "Content-Type: application/json" \
-		-d "{\"username\":\"$PRISMA_ACCESS_KEY\",\"password\":\"$PRISMA_SECRET\"}" \
-		"${API_URL}/login")
+	local request_data=$(
+		cat <<- END
+		{
+			"password": "${LIFERAY_PRISMA_SECRET}",
+			"username": "${LIFERAY_PRISMA_ACCESS_KEY}"
+		}
+		END
+	)
 
-	TOKEN=$(echo "$auth_response" | jq -r '.token')
+	local auth_response=$(\
+		curl \
+			--data "${request_data}" \
+			--header "Content-Type: application/json" \
+			--request POST \
+			--silent \
+			"${api_url}/login")
+
+	local token=$(echo "${auth_response}" | jq -r '.token')
+
+	curl \
+		--header "x-redlock-auth: ${token}" \
+		--output twistcli \
+		--silent \
+		"${console}/api/v1/util/twistcli"
 
-	# Download twistcli
-	curl -s -H "x-redlock-auth: $TOKEN" -o twistcli "${CONSOLE}/api/v1/util/twistcli"
 	chmod +x ./twistcli
 
-	# Scan each image
-	for image_name in "${image_names[@]}"; do
-		local sanitized_image_name=$(echo "$image_name" | sed 's/[^a-zA-Z0-9]/_/g')
+	for image_name in "${image_names[@]}"
+	do
+		local sanitized_image_name=$(echo "${image_name}" | sed 's/[^a-zA-Z0-9]/_/g')
+
 		local vulnerabilities_file="vulnerabilities_${sanitized_image_name}.json"
 
-		# Scan the image
-		scan_output=$(./twistcli images scan \
-			--address "$CONSOLE" \
-			--docker-address "/run/user/1000/docker.sock" \
-			--user "$PRISMA_ACCESS_KEY" \
-			--password "$PRISMA_SECRET" \
-			"$image_name")
+		lc_log INFO "Scanning ${image_name}."
+
+		export scan_output=$(\
+			./twistcli images scan \
+				--address "${console}" \
+				--docker-address "/run/user/1000/docker.sock" \
+				--password "${LIFERAY_PRISMA_SECRET}" \
+				--user "${LIFERAY_PRISMA_ACCESS_KEY}" \
+				"${image_name}")
+
+		lc_log INFO "Scan output for ${image_name}:"
+
+		lc_log INFO "${scan_output}"
+
+		if [[ ${scan_output} == *"Vulnerability threshold check results: PASS"* ]] || [[ ${image_name} == *"7.4.13.nightly"* ]]
+		then
+			lc_log INFO "The result of scan for ${image_name} is: PASS"
+		else
+			lc_log INFO "The result of scan for ${image_name} is: FAIL"
 
-		echo "Scan output for $image_name:"
-		echo "$scan_output"
+			lc_log ERROR "The Docker image ${image_name} has security vulnerabilities."
+		fi
```

Comment:
@lucasmiranda0, please remove the nightly check and clarify with William Azevedo the expected behavior for this code snippet.